xbmc/addons/PVRClient.cpp

```diff
@@ -657,6 +689,46 @@ PVR_ERROR CPVRClient::RenameChannel(const CPVRChannelPtr &channel)
   return retVal;
 }
 
+PVR_ERROR CPVRClient::IsRecordable(const EPG::CConstEpgInfoTagPtr &tag, bool *isRecordable)
+{
+  if (!m_bReadyToUse)
+    return PVR_ERROR_SERVER_ERROR;
+
+  if (!m_addonCapabilities.bSupportsRecordings)
+    return PVR_ERROR_NOT_IMPLEMENTED;
+
+  PVR_ERROR retVal(PVR_ERROR_UNKNOWN);
+
+  EPG_TAG pvrTag;
+  WriteEpgTag(tag, pvrTag);
```

Comment:
As far as I understand, Kodi does cache the EPG. I expect that most add-ons do not cache this information. Also, I expect that It is possible that this function is called with an EPG entry which is not in the cache of the add-on (e.g. if Kodi used the information form its cache on did not request a refresh from the add-on).
In my case (Zattoo) it would be possible to request the EPG information from their website, but that would be far more expensive. Also I'm not sure if that will be possible for all other providers.
This is why I decided to pass the full entry back to the addon. Do you still suggest to only pass the id? If yes, I will change the API.